### PR TITLE
Add filter UI and filter by content types

### PR DESCRIPTION
**Pull request by AdmiralCoco**
(Filtering by dates currently does not work but is in UI)

### CODE REVIEW

**Comment by AdmiralCoco** [comment]
image

image

File: src/components/content-list/content-list.js [hunk 1]
```diff
@@ -125,21 +148,16 @@ class ContentList extends DependencyRequester(InternalLocalizeMixin(NavigationMi
...
 
 	async reloadPage() {
 		this.loading = true;
 		this.contentItems = [];
 		this.searchQueryStart = 0;
-		this._navigate('/manage/content', {
-			q: encodeURIComponent(this.searchQuery),
-			sortQuery: this.sortQuery
-		});
+		this._navigate('/manage/content', this.queryParams);
```
Comment (qwangd2l) [inline src/components/content-list/content-list.js]: Do we still need to encode `this.queryParams.searchQuery` like it was doing previously?

File: src/components/content-list/content-list.js [hunk 2]
```diff
@@ -60,9 +59,15 @@ class ContentList extends DependencyRequester(InternalLocalizeMixin(NavigationMi
...
-		this.searchQuery = q;
+		const {
+			searchQuery = '',
+			sortQuery = 'updatedAt:desc',
+			contentType = '',
+			dateCreated = '',
+			dateModified = ''
+		} = rootStore.routingStore.getQueryParams();
+
+		this.queryParams = { searchQuery, sortQuery, contentType, dateCreated, dateModified };
```
Comment (qwangd2l) [inline src/components/content-list/content-list.js]: I think there are a couple of places from my PR today that was using `this.sortQuery`. I can try to fix this after you merge this or you can try to replace their references in this file.

File: src/components/content-list/content-list.js (hunk 1 above)
Comment (qwangd2l) [reply src/components/content-list/content-list.js]: Oh I see it encoded in the navigation mixin.

File: src/pages/content-page.js [hunk 3]
```diff
@@ -23,12 +24,11 @@ class ContentPage extends PageViewElement {
...
-
-			@media (max-width: 900px) {
-				.content-page-header {
-					align-items: flex-start;
-					flex-direction: column;
-				}
+			.content-page-search-filter-section {
+				display: flex;
+			}
+			content-filter-dropdown {
```
Comment (qwangd2l) [inline src/pages/content-page.js]: I think we'll need a RTL mixin for this.

File: src/pages/content-page.js [hunk 4]
```diff
@@ -23,12 +24,11 @@ class ContentPage extends PageViewElement {
...
+				padding-right: 1rem;
```
Comment (qwangd2l) [inline src/pages/content-page.js]: I think we'll need a RTL mixin for this.

File: src/pages/content-page.js [hunk 5]
```diff
@@ -30,6 +30,9 @@ class ContentPage extends PageViewElement {
 			content-filter-dropdown {
 				padding-right: 1rem;
 			}
+			:host([dir="rtl"]) content-filter-dropdown {
+				padding-left: 1rem;
```
Comment (MichaelKovalchuk) [inline src/pages/content-page.js]: my understanding that the content-filter-dropdown will have right and left padding set to 1rem in rtl, right? I may be wrong in how the rtl is treated by the browser.

Comment (qwangd2l) [reply src/pages/content-page.js]: Yeah, that's how it works. 
I believe we need `padding-right: 0;` in `:host([dir="rtl"]) content-filter-dropdown` as well.

Comment (AdmiralCoco) [reply src/pages/content-page.js]: Oops yep you guys are right - will update, thanks

**Pull request merged**